File: 1510/order.cpp
#include <stdio.h>
#include <map>

using namespace std;

int main() {
	map<int, int> counts;
	int max_count = 0;
	int max_num = 0;
	int i, n, t, x;

	scanf("%d", &n);

	for(i=0; i<n; i++)
	{
		scanf("%d", &t);

		x = counts[t] + 1;
		counts[t] = x;

		if(x > max_count)
		{
			max_count = x;
			max_num = t;
		}
	}

	printf("%d\n", max_num);
	return 0;
}

File: 1510/order.py
n = int(input().strip())

counter = {}
max_count = 0
max_num = 0

for i in range(n):
	t = int(input().strip())

	try:
		counter[t] = counter[t] + 1
	except KeyError:
		counter[t] = 1

	if max_count < counter[t]:
		max_count = counter[t]
		max_num = t

print(max_num)